recipes/agg/all/conanfile.py

```diff
@@ -0,0 +1,140 @@
+from conan import ConanFile
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.files import copy, get, rmdir
+from conan.errors import ConanInvalidConfiguration
+
+import os
+
+required_conan_version = ">=1.52.0"
+
+class AggConan(ConanFile):
+    name = 'agg'
+    description = 'AGG Anti-Grain Geometry Library'
+    topics = ('graphics')
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = 'https://github.com/aggeom'
+    license = 'BSD-3-Clause'
+
+    settings = 'os', 'arch', 'compiler', 'build_type'
+    options = {
+        'shared': [True, False],
+        'fPIC': [True, False],
+        'with_gpc': [True, False],
+        'with_freetype': [True, False],
+        'with_agg2d': [True, False],
+        'with_agg2d_freetype': [True, False],
+        'with_platform': [True, False],
+        'with_controls': [True, False],
+    }
+
+    default_options = {
+        'shared': False,
+        'fPIC': True,
+        'with_gpc': True,
+        'with_freetype': True,
+        'with_agg2d': True,
+        'with_agg2d_freetype': True,
+        'with_platform': True,
+        'with_controls': True,
+    }
+
+    def validate(self):
+        if self.settings.os not in ("Windows", "Linux"):
+            raise ConanInvalidConfiguration("OS is not supported")
+        if self.options.shared:
+            raise ConanInvalidConfiguration("Invalid configuration")
```

Comment:
Is there a reason to have this option? can we remove it?

Reply:
Maybe you are right. I still leave it there because maybe someone will test shared library, and we'll enable it.

Reply:
@Jihadist please, sign the CLA again. It has been updated. Sorry for the inconvenient.

Reply:
@uilianries thank you, I signed